### ISSUE TITLE
`textDocument/definition` not working in nvim

### ISSUE THREAD

**Issue by cinghiopinghio** [muffinmad/anakin-language-server#17]
Thanks for the great project.

While experimenting with builtin LSP in nvim I found that goto definition do not work with `anakinls` as language server (it doesn't work with `jedi-language-server` either).
Unfotunately `anakinls` do not work with `LanguageClient` either with the following error message:
```
[LC] data did not match any variant of untagged enum GotoDefinitionResponse
```

I opened an issue at `nvim-lspconfig` at https://github.com/neovim/nvim-lspconfig/issues/424  but they pointed out that it may be a upstream bug (They suggest that the feature is not implemented but my understanding from the `readme` is that that feature should be implemented).

Please find here the relevant log relative to the nvim builtin LSP:
```
[ DEBUG ] 2020-11-24T10:02:57+0100 ] /usr/share/nvim/runtime/lua/vim/lsp.lua:670 ]      "LSP[anakin_language_server]"   "client.request"        1       "textDocument/definition"       {  position = {    character = 13,    line = 622  },  textDocument = {    uri = "file:///home/xxxx/codes/mycode/mycode/base.py"  }}        <function 1>    1
[ DEBUG ] 2020-11-24T10:02:57+0100 ] /usr/share/nvim/runtime/lua/vim/lsp/rpc.lua:376 ]  "rpc.send.payload"      {  id = 2,  jsonrpc = "2.0",  method = "textDocument/definition",  params = {    position = {      character = 13,      line = 622    },    textDocument = {      uri = "file:///home/xxxx/codes/mycode/mycode[ DEBUG ] 2020-11-24T10:02:57+0100 ] /usr/share/nvim/runtime/lua/vim/lsp.lua:670 ]      "LSP[anakin_language_server]"   "client.request"        1       "textDocument/definition"       {  position = {    character = 13,    line = 622  },  textDocument = {    uri = "file:///home/xxxx/codes/mycode/mycode/base.py"  }}        <function 1>    1
[ DEBUG ] 2020-11-24T10:02:57+0100 ] /usr/share/nvim/runtime/lua/vim/lsp/rpc.lua:376 ]  "rpc.send.payload"      {  id = 2,  jsonrpc = "2.0",  method = "textDocument/definition",  params = {    position = {      character = 13,      line = 622    },    textDocument = {      uri = "file:///home/xxxx/codes/mycode/mycode/base.py"    }  }}
[ DEBUG ] 2020-11-24T10:02:57+0100 ] /usr/share/nvim/runtime/lua/vim/lsp/rpc.lua:477 ]  "decoded"       {  id = 2,  jsonrpc = "2.0",  result = { {      range = {        end = {          character = 12,          line = 635        },        start = {          character = 4,          line = 635        }      },      uri = vim.NIL    } }}
[ DEBUG ] 2020-11-24T10:02:57+0100 ] /usr/share/nvim/runtime/lua/vim/lsp/handlers.lua:297 ]     "default_handler"       "textDocument/definition"       {  bufnr = 1,  client_id = 1,  params = { {      range = {        end = {          character = 12,          line = 635        },        start = {          character = 4,          line = 635        }      }    } }}
[ INFO ] 2020-11-24T10:03:01+0100 ] /usr/share/nvim/runtime/lua/vim/lsp.lua:974 ]       "exit_handler"  { {    _on_attach = <function 1>,    callbacks = <1>{      ["window/logMessage"] = <function 2>,      ["window/showMessage"] = <function 3>,      ["workspace/configuration"] = <function 4>    },    cancel_request = <function 5>,    config = {      _on_attach = <function 6>,      capabilities = {        callHierarchy = {          dynamicRegistration = false,          <metatable> = <2>{            __tostring = <function 7>          }        },        textDocument = {          codeAction = {            codeActionLiteralSupport = {              codeActionKind = {                valueSet = { "", "RefactorInline", "source.organizeImports", "quickfix", "QuickFix", "Refactor", "Source", "RefactorRewrite", "refactor.inline", "RefactorExtract", "source", "SourceOrganizeImports", "refactor", "Empty", "refactor.rewrite", "refactor.extract" },                <metatable> = <table 2>              },              <metatable> = <table 2>            },            dynamicRegistration = false,            <metatable> = <table 2>          },          completion = {            completionItem = {              commitCharactersSupport = false,              deprecatedSupport = false,              documentationFormat = { "markdown", "plaintext" },              preselectSupport = false,              snippetSupport = false,              <metatable> = <table 2>            },            completionItemKind = {              valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25 },              <metatable> = <table 2>            },            contextSupport = false,            dynamicRegistration = false,            <metatable> = <table 2>          },          declaration = {            linkSupport = true,            <metatable> = <table 2>          },          definition = {            linkSupport = true,            <metatable> = <table 2>          },          documentHighlight = {            dynamicRegistration = false,            <metatable> = <table 2>          },          documentSymbol = {            dynamicRegistration = false,            hierarchicalDocumentSymbolSupport = true,            symbolKind = {              valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26 },              <metatable> = <table 2>            },            <metatable> = <table 2>          },          hover = {            contentFormat = { "markdown", "plaintext" },            dynamicRegistration = false,            <metatable> = <table 2>          },          implementation = {            linkSupport = true,            <metatable> = <table 2>          },          references = {            dynamicRegistration = false,            <metatable> = <table 2>          },          rename = {            dynamicRegistration = false,            prepareSupport = true,            <metatable> = <table 2>          },          signatureHelp = {            dynamicRegistration = false,            signatureInformation = {              documentationFormat = { "markdown", "plaintext" },              <metatable> = <table 2>            },            <metatable> = <table 2>          },          synchronization = {            didSave = true,            dynamicRegistration = false,            willSave = false,            willSaveWaitUntil = false,            <metatable> = <table 2>          },          typeDefinition = {            linkSupport = true,            <metatable> = <table 2>          },          <metatable> = <table 2>        },        workspace = {          applyEdit = true,          configuration = true,          symbol = {            dynamicRegistration = false,            hierarchicalWorkspaceSymbolSupport = true,            symbolKind = {              valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26 },              <metatable> = <table 2>            },            <metatable> = <table 2>          },          <metatable> = <table 2>        }      },      cmd = { "anakinls" },      filetypes = { "python" },      handlers = <table 1>,      init_options = vim.empty_dict(),      log_level = 2,      message_level = 2,      name = "anakin_language_server",      on_attach = <function 8>,      on_exit = <function 9>,      on_init = <function 10>,      root_dir = "/home/xxxx/codes/mycode/mycode",      settings = vim.empty_dict(),      <metatable> = <table 2>    },    handlers = <table 1>,    id = 1,    initialized = true,    is_stopped = <function 11>,    name = "anakin_language_server",    notify = <function 12>,    offset_encoding = "utf-16",    request = <function 13>,    resolved_capabilities = {      call_hierarchy = false,      code_action = <3>{        codeActionKinds = { "refactor.inline", "refactor.extract" }      },      completion = true,      declaration = false,      document_formatting = true,      document_highlight = true,      document_range_formatting = true,      document_symbol = true,      execute_command = true,      find_references = true,      goto_definition = true,      hover = true,      implementation = false,      rename = true,      signature_help = true,      signature_help_trigger_characters = <4>{ "(", "," },      text_document_did_change = 2,      text_document_open_close = true,      text_document_save = <5>{        includeText = false      },      text_document_save_include_text = false,      text_document_will_save = false,      text_document_will_save_wait_until = false,      type_definition = false,      workspace_symbol = false    },    rpc = {      handle = <userdata 1>,      notify = <function 14>,      pid = 547204,      request = <function 15>    },    server_capabilities = {      codeActionProvider = <table 3>,      completionProvider = {        resolveProvider = false,        triggerCharacters = { "." }      },      definitionProvider = true,      documentFormattingProvider = true,      documentHighlightProvider = true,      documentOnTypeFormattingProvider = {        firstTriggerCharacter = "",        moreTriggerCharacter = {}      },      documentRangeFormattingProvider = true,      documentSymbolProvider = true,      executeCommandProvider = {        commands = {}      },      hoverProvider = true,      referencesProvider = true,      renameProvider = true,      signatureHelpProvider = {        triggerCharacters = <table 4>      },      textDocumentSync = {        change = 2,        openClose = true,        save = <table 5>,        willSave = false,        willSaveWaitUntil = false      },      workspace = {        workspaceFolders = {          changeNotifications = true,          supported = true        }      },      workspaceSymbolProvider = false    },    stop = <function 16>,    supports_method = <function 17>,    workspace_did_change_configuration = <function 18>  } }
/base.py"    }  }}
[ DEBUG ] 2020-11-24T10:02:57+0100 ] /usr/share/nvim/runtime/lua/vim/lsp/rpc.lua:477 ]  "decoded"       {  id = 2,  jsonrpc = "2.0",  result = { {      range = {        end = {          character = 12,          line = 635        },        start = {          character = 4,          line = 635        }      },      uri = vim.NIL    } }}
[ DEBUG ] 2020-11-24T10:02:57+0100 ] /usr/share/nvim/runtime/lua/vim/lsp/handlers.lua:297 ]     "default_handler"       "textDocument/definition"       {  bufnr = 1,  client_id = 1,  params = { {      range = {        end = {          character = 12,          line = 635        },        start = {          character = 4,          line = 635        }      }    } }}
[ INFO ] 2020-11-24T10:03:01+0100 ] /usr/share/nvim/runtime/lua/vim/lsp.lua:974 ]       "exit_handler"  { {    _on_attach = <function 1>,    callbacks = <1>{      ["window/logMessage"] = <function 2>,      ["window/showMessage"] = <function 3>,      ["workspace/configuration"] = <function 4>    },    cancel_request = <function 5>,    config = {      _on_attach = <function 6>,      capabilities = {        callHierarchy = {          dynamicRegistration = false,          <metatable> = <2>{            __tostring = <function 7>          }        },        textDocument = {          codeAction = {            codeActionLiteralSupport = {              codeActionKind = {                valueSet = { "", "RefactorInline", "source.organizeImports", "quickfix", "QuickFix", "Refactor", "Source", "RefactorRewrite", "refactor.inline", "RefactorExtract", "source", "SourceOrganizeImports", "refactor", "Empty", "refactor.rewrite", "refactor.extract" },                <metatable> = <table 2>              },              <metatable> = <table 2>            },            dynamicRegistration = false,            <metatable> = <table 2>          },          completion = {            completionItem = {              commitCharactersSupport = false,              deprecatedSupport = false,              documentationFormat = { "markdown", "plaintext" },              preselectSupport = false,              snippetSupport = false,              <metatable> = <table 2>            },            completionItemKind = {              valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25 },              <metatable> = <table 2>            },            contextSupport = false,            dynamicRegistration = false,            <metatable> = <table 2>          },          declaration = {            linkSupport = true,            <metatable> = <table 2>          },          definition = {            linkSupport = true,            <metatable> = <table 2>          },          documentHighlight = {            dynamicRegistration = false,            <metatable> = <table 2>          },          documentSymbol = {            dynamicRegistration = false,            hierarchicalDocumentSymbolSupport = true,            symbolKind = {              valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26 },              <metatable> = <table 2>            },            <metatable> = <table 2>          },          hover = {            contentFormat = { "markdown", "plaintext" },            dynamicRegistration = false,            <metatable> = <table 2>          },          implementation = {            linkSupport = true,            <metatable> = <table 2>          },          references = {            dynamicRegistration = false,            <metatable> = <table 2>          },          rename = {            dynamicRegistration = false,            prepareSupport = true,            <metatable> = <table 2>          },          signatureHelp = {            dynamicRegistration = false,            signatureInformation = {              documentationFormat = { "markdown", "plaintext" },              <metatable> = <table 2>            },            <metatable> = <table 2>          },          synchronization = {            didSave = true,            dynamicRegistration = false,            willSave = false,            willSaveWaitUntil = false,            <metatable> = <table 2>          },          typeDefinition = {            linkSupport = true,            <metatable> = <table 2>          },          <metatable> = <table 2>        },        workspace = {          applyEdit = true,          configuration = true,          symbol = {            dynamicRegistration = false,            hierarchicalWorkspaceSymbolSupport = true,            symbolKind = {              valueSet = { 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26 },              <metatable> = <table 2>            },            <metatable> = <table 2>          },          <metatable> = <table 2>        }      },      cmd = { "anakinls" },      filetypes = { "python" },      handlers = <table 1>,      init_options = vim.empty_dict(),      log_level = 2,      message_level = 2,      name = "anakin_language_server",      on_attach = <function 8>,      on_exit = <function 9>,      on_init = <function 10>,      root_dir = "/home/xxxx/codes/mycode/mycode",      settings = vim.empty_dict(),      <metatable> = <table 2>    },    handlers = <table 1>,    id = 1,    initialized = true,    is_stopped = <function 11>,    name = "anakin_language_server",    notify = <function 12>,    offset_encoding = "utf-16",    request = <function 13>,    resolved_capabilities = {      call_hierarchy = false,      code_action = <3>{        codeActionKinds = { "refactor.inline", "refactor.extract" }      },      completion = true,      declaration = false,      document_formatting = true,      document_highlight = true,      document_range_formatting = true,      document_symbol = true,      execute_command = true,      find_references = true,      goto_definition = true,      hover = true,      implementation = false,      rename = true,      signature_help = true,      signature_help_trigger_characters = <4>{ "(", "," },      text_document_did_change = 2,      text_document_open_close = true,      text_document_save = <5>{        includeText = false      },      text_document_save_include_text = false,      text_document_will_save = false,      text_document_will_save_wait_until = false,      type_definition = false,      workspace_symbol = false    },    rpc = {      handle = <userdata 1>,      notify = <function 14>,      pid = 547204,      request = <function 15>    },    server_capabilities = {      codeActionProvider = <table 3>,      completionProvider = {        resolveProvider = false,        triggerCharacters = { "." }      },      definitionProvider = true,      documentFormattingProvider = true,      documentHighlightProvider = true,      documentOnTypeFormattingProvider = {        firstTriggerCharacter = "",        moreTriggerCharacter = {}      },      documentRangeFormattingProvider = true,      documentSymbolProvider = true,      executeCommandProvider = {        commands = {}      },      hoverProvider = true,      referencesProvider = true,      renameProvider = true,      signatureHelpProvider = {        triggerCharacters = <table 4>      },      textDocumentSync = {        change = 2,        openClose = true,        save = <table 5>,        willSave = false,        willSaveWaitUntil = false      },      workspace = {        workspaceFolders = {          changeNotifications = true,          supported = true        }      },      workspaceSymbolProvider = false    },    stop = <function 16>,    supports_method = <function 17>,    workspace_did_change_configuration = <function 18>  } }
```

**Comment by muffinmad** [comment]
Can you please provide some step-by-step receipt for reproducing this?

What I've checked:

`~/.config/nvim/init.vim` file:
```
call plug#begin('~/.vim/plugged')
Plug 'autozimu/LanguageClient-neovim', {
    \ 'branch': 'next',
    \ 'do': 'bash install.sh',
    \ }
call plug#end()

let g:LanguageClient_serverCommands = {
                        \ 'python': ['tcp://127.0.0.1:2087'],
                        \}
nnoremap <silent> gd :call LanguageClient#textDocument_definition()<CR>
```

The server is running by the command `python3 -m pdb -m anakinls --tcp`.

`~/test.py` file:
```python
def foo():
    pass


foo
```

With the point on the last line typing `gd` move the point to the `def foo`.

**Comment by cinghiopinghio** [comment]
Your minimal init is not working for me: i get `[LC] Language server is not running for: python`.

Changing the following:
```
let g:LanguageClient_serverCommands = {
                        \ 'python': ['anakinls'],
                        \}
```
when pressing `gd` I still have:
```
[LC] data did not match any variant of untagged enum GotoDefinitionResponse
```
diagnostics are working as expected.

I don't know how to investigate this using the pdb console though.

**Comment by muffinmad** [comment]
> Changing the following:
> ```
> let g:LanguageClient_serverCommands = {
>                         \ 'python': ['anakinls'],
>                         \}
> ```
> when pressing `gd` I still have:
> ```
> [LC] data did not match any variant of untagged enum GotoDefinitionResponse
> ```

Does this happens in that `test.py` file?

**Comment by cinghiopinghio** [comment]
Yes, the error is spawned by `nvim` when I press `gd` and the cursor is on the call of the function `foo` (last line of `test.py`).

**Comment by muffinmad** [comment]
Just checked under ArchLinux with `pacman -S neovim` and `pip install --user anakin-language-server` and the same init file. `gd` works as expected.

```
NVIM v0.4.4
Build type: Release
LuaJIT 2.0.5
Compilation: /usr/bin/cc -D_FORTIFY_SOURCE=2 -march=x86-64 -mtune=generic -O2 -pipe -fno-plt -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1 -O2 -DNDEBUG -DMIN_LOG_LEVEL=3 -Wall -Wextra -pedantic -Wno-unused-parameter -Wstrict-prototypes -std=gnu99 -Wshadow -Wconversion -Wmissing-prototypes -Wimplicit-fallthrough -Wvla -fstack-protector-strong -fno-common -fdiagnostics-color=always -DINCLUDE_GENERATED_DECLARATIONS -D_GNU_SOURCE -DNVIM_MSGPACK_HAS_FLOAT32 -DNVIM_UNIBI_HAS_VAR_FROM -I/build/neovim/src/build/config -I/build/neovim/src/neovim-0.4.4/src -I/usr/include -I/build/neovim/src/build/src/nvim/auto -I/build/neovim/src/build/include
Compiled by builduser

Features: +acl +iconv +tui
```

**Comment by cinghiopinghio** [comment]
Whaat?

I'm in the very same environment:
- Archlinux
- nvim (tested both `neovim` and `neovim-git`)
- pip-installed `anakin-language-server`

No luck so far.

**Comment by muffinmad** [comment]
I've added following lines to my `init.vim`:
```vim
let g:LanguageClient_trace = 'verbose'
let g:LanguageClient_loggingFile = expand('~/LanguageClient.log')
let g:LanguageClient_loggingLevel = 'DEBUG'
let g:LanguageClient_serverStderr = expand('~/LanguageClientStderr.log')
```

Inspecting the `~/LanguageClientStderr.log` I've spotted exception in the server initialization code due to missing `initializationOptions` param. It must be fixed in https://github.com/muffinmad/anakin-language-server/releases/tag/v1.10.1.

Don't know why the behavior is different in nearly identical setup though.

Please update `anakin-language-server` to the latest version, enable logging as above and let's look further.

**Comment by cinghiopinghio** [comment]
Unfortunately this do not solve the problem. `LanguageClientStderr.log` is now clean (only INFOs rows).

On the other hand in `LanguageClient.log` I have:

```
14:46:09 DEBUG writer-Some("python") src/rpcclient.rs:254 => Some("python") {"jsonrpc":"2.0","method":"textDocument/definition","params":{"bufnr":1,"character":0,"filename":"/home/xxxx/test.py","gotoCmd":null,"handle":true,"languageId":"python","line":4,"method":"textDocument/definition","position":{"character":0,"line":4},"text":["def foo():","    print('pippo')","","","foo()","faaa()",""],"textDocument":{"uri":"file:///home/xxxx/test.py"}},"id":1}
14:46:09 DEBUG reader-Some("python") src/rpcclient.rs:207 <= Some("python") {"id": 1, "jsonrpc": "2.0", "result": [{"uri": null, "range": {"start": {"line": 0, "character": 4}, "end": {"line": 0, "character": 7}}}]}
14:46:09 ERROR unnamed src/rpchandler.rs:27 Error handling message: data did not match any variant of untagged enum GotoDefinitionResponse
Message: {"jsonrpc":"2.0","method":"textDocument/definition","params":{"bufnr":1,"character":0,"filename":"/home/xxxx/test.py","gotoCmd":null,"handle":true,"languageId":"python","line":4,"text":["def foo():","    print('pippo')","","","foo()","faaa()",""]},"id":1}
Error: data did not match any variant of untagged enum GotoDefinitionResponse
14:46:09 DEBUG writer-None src/rpcclient.rs:254 => None {"jsonrpc":"2.0","error":{"code":-32603,"message":"data did not match any variant of untagged enum GotoDefinitionResponse"},"id":1}
```

I'm a bit ignorant about what exactly each of those lines means but I can guess.
When testing against `pyls` I notice that the url returned (line `reader-Some("python")`) is `null` while in `pyls` the path to the file is returned `file:///path/to/file.py`.

Do you have a different log?

**Comment by muffinmad** [comment]
> Do you have a different log?

Yes. In my case `uri` is properly set:

```
16:04:40 DEBUG writer-Some("python") src/rpcclient.rs:254 => Some("python") {"jsonrpc":"2.0","method":"textDocument/definition","params":{"bufnr":1,"character":0,"filename":"/home/mad/test.py","gotoCmd":null,"handle":true,"languageId":"python","line":4,"method":"textDocument/definition","position":{"character":0,"line":4},"text":["def foo():","    pass","","","foo",""],"textDocument":{"uri":"file:///home/mad/test.py"}},"id":1}
16:04:40 DEBUG reader-Some("python") src/rpcclient.rs:207 <= Some("python") {"id": 1, "jsonrpc": "2.0", "result": [{"uri": "file:///home/mad/test.py", "range": {"start": {"line": 0, "character": 4}, "end": {"line": 0, "character": 7}}}]}
```

What python is used by `anakinls`? Content of my `LanguageClientStderr.log`:

```
INFO:pygls.server:Starting IO server
INFO:root:Jedi environment python: /usr/bin/python
INFO:root:Jedi environment sys_path:
INFO:root:  /usr/lib/python39.zip
INFO:root:  /usr/lib/python3.9
INFO:root:  /usr/lib/python3.9/lib-dynload
INFO:root:  /home/mad/.local/lib/python3.9/site-packages
INFO:root:  /usr/lib/python3.9/site-packages
INFO:root:Jedi project path: /home/mad
```

**Comment by cinghiopinghio** [comment]
I have the same log except that it also contains:
```
INFO:root:  /usr/lib/python3.9/site-packages/linkgrammar
```

**Comment by cinghiopinghio** [comment]
Ah, I also use `python-jedi-git` since I had other issues with `v0.17.2-3`.

**Comment by cinghiopinghio** [comment]
Also, in method `textDocument/publishDiagnostics`, the uri is properly passed back to the client.

**Comment by muffinmad** [comment]
> Ah, I also use `python-jedi-git` since I had other issues with `v0.17.2-3`.

At last! :) I can reproduce you issue after `pip install -U --user git+https://github.com/davidhalter/jedi.git`

**Comment by muffinmad** [comment]
With latest stable `jedi` version:

```
$ pip install jedi
$ python
Python 3.7.9 (default, Sep  9 2020, 00:09:07)
[Clang 11.0.3 (clang-1103.0.32.62)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from jedi import Script
>>> script=Script(code='def foo():\n    pass\nfoo', path='/home/test/test.py')
>>> definitions=script.goto(3, 0)
>>> definitions[0].module_path
'/home/test/test.py'
```

With latest development `jedi` version:

```
$ pip install -U git+https://github.com/davidhalter/jedi.git
$ python
Python 3.7.9 (default, Sep  9 2020, 00:09:07)
[Clang 11.0.3 (clang-1103.0.32.62)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from jedi import Script
>>> script=Script(code='def foo():\n    pass\nfoo', path='/home/test/test.py')
>>> definitions=script.goto(3, 0)
>>> definitions[0].module_path
PosixPath('/home/test/test.py')
```

@davidhalter can you please give some comments about such behavior? Or it's the same as https://github.com/davidhalter/jedi/issues/1661#issuecomment-683315478? TIA!

**Comment by davidhalter** [comment]
This is intentional, yes. It's also part of the Changelog.

I know this is kind of breaking the API, but that's just because I'm dropping Python 2 in 0.18.0 and wanted to clean up some API details.

**Comment by muffinmad** [comment]
@cinghiopinghio Must be fixed in `1.10.2`

**Comment by cinghiopinghio** [comment]
I confirm that it's now working as expected. Thanks.
